realm/kotlin-extensions/src/main/kotlin/io/realm/kotlin/RealmObjectExtensions.kt

```diff
@@ -56,94 +54,25 @@ import kotlinx.coroutines.flow.flowOf
-            return@callbackFlow
-        }
-
-        // Get instance to ensure the Realm is open for as long as we are listening
-        val flowRealm = Realm.getInstance(config)
-        val listener = RealmChangeListener<T> { listenerObj ->
-            offer(listenerObj.freeze())
-        }
-
-        obj.addChangeListener(listener)
-
-        // Emit current (frozen) value
-        offer(freeze(obj))
-
-        awaitClose {
-            // Remove listener and cleanup
-            if (!flowRealm.isClosed) {
-                obj.removeChangeListener(listener)
-                flowRealm.close()
-            }
-        }
-    }
-}
-
-private fun flowFromDynamicRealm(
-        dynamicRealm: DynamicRealm,
-        dynamicObject: DynamicRealmObject
-): Flow<DynamicRealmObject> {
-    // Return "as is" if frozen, there will be no listening for changes
-    if (dynamicRealm.isFrozen) {
-        return flowOf(dynamicObject)
-    }
-
-    val config = dynamicRealm.configuration
-
-    return callbackFlow<DynamicRealmObject> {
-        // Do nothing if the object is invalid
-        if (!dynamicObject.isValid()) {
-            return@callbackFlow
-        }
-
-        // Get instance to ensure the Realm is open for as long as we are listening
-        val flowRealm = Realm.getInstance(config)
-        val listener = RealmChangeListener<DynamicRealmObject> { listenerObj ->
-            offer(listenerObj.freeze())
-        }
-
-        dynamicObject.addChangeListener(listener)
-
-        // Emit current (frozen) value
-        offer(freeze(dynamicObject))
-
-        awaitClose {
-            // Remove listener and cleanup
-            if (!flowRealm.isClosed) {
-                dynamicObject.removeChangeListener(listener)
-                flowRealm.close()
+fun <T : RealmModel> T?.toFlow(): Flow<T?> {
+    // Return flow with object or null flow if this function is called on null
+    return this?.let { obj ->
+        if (obj is RealmObjectProxy) {
+            val proxy = obj as RealmObjectProxy
+
+            @Suppress("INACCESSIBLE_TYPE")
+            when (val realm = proxy.`realmGet$proxyState`().`realm$realm`) {
+                is Realm -> realm.configuration.flowFactory?.from<T>(realm, obj)
+                        ?: throw IllegalStateException("Missing flow factory in Realm configuration.")
+                is DynamicRealm ->
+                    (obj as DynamicRealmObject).let { dynamicRealmObject ->
+                        (realm.configuration.flowFactory?.from(realm, dynamicRealmObject)
+                                ?: throw IllegalStateException("Missing flow factory in Realm configuration.")) as Flow<T?>
+                    }
+                else -> throw UnsupportedOperationException("${realm.javaClass} is not supported as a candidate for 'toFlow'. Only subclasses of RealmModel/RealmObject can be used.")
             }
+        } else {
+            return flowOf(this)
         }
-    }
+    } ?: flowOf(null)
```

Comment:
We seem to be missing tests for this?

Reply:
No, it is here `CoroutinesTests.realmObject_toFlow_nullObjectEmitsNullFlow`.